src/hpack/struct.py

```diff
@@ -37,3 +39,10 @@ class NeverIndexedHeaderTuple(HeaderTuple):
     __slots__ = ()
 
     indexable = False
+
+    def __new__(cls, *args: Any) -> "NeverIndexedHeaderTuple":
+        return tuple.__new__(cls, args)
+
+
+# explicitly mentioning all valid header types , even if some superseed each other
+type Headers = Iterable[Union[HeaderTuple, NeverIndexedHeaderTuple, tuple[Union[bytes, str], Union[bytes, str]]]]
```

Comment:
Better not to allow mixed `bytes` and `str` headers as we don't expect them.

```suggestion
type Headers = Iterable[Union[HeaderTuple, NeverIndexedHeaderTuple, tuple[bytes, bytes], tuple[str, str]]]
```

Reply:
I had that originally but then made it a bit softer to accommodate other typing issues in the h2 project ... I'll try to make it stricter again, as suggested here.